subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/internal/variantreports/formatter/ConsoleVariantReportWriter.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks.diagnostics.internal.variantreports.formatter;
+
+import org.apache.commons.lang.StringUtils;
+import org.gradle.api.tasks.diagnostics.internal.variantreports.model.ReportArtifact;
+import org.gradle.api.tasks.diagnostics.internal.variantreports.model.ReportAttribute;
+import org.gradle.api.tasks.diagnostics.internal.variantreports.model.ReportCapability;
+import org.gradle.api.tasks.diagnostics.internal.variantreports.model.ReportConfiguration;
+import org.gradle.api.tasks.diagnostics.internal.variantreports.model.ReportSecondaryVariant;
+import org.gradle.internal.logging.text.StyledTextOutput;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public final class ConsoleVariantReportWriter implements VariantReportWriter {
+    private final StyledTextOutput output;
+    private int depth;
+
+    private final String projectName;
+    private final String targetName;
+    private final String direction;
+    private final boolean includeCapabilities;
+    private final boolean includeArtifacts;
+    private final boolean includeSecondaryVariants;
```

Comment:
These almost feel overboard since they're all either true or false right now.

Reply:
Yeah, I thought at first these might vary more, but I think now they can all be summed up by which direction we're reporting.  I think I'll make that an enum that holds all the target, direction + inclusion info.